cwms-data-api/src/main/java/cwms/cda/data/dao/StreamLocationDao.java

```diff
@@ -0,0 +1,184 @@
+                    streamLocation.getUngagedDrainageArea(), streamLocation.getAreaUnits(),streamLocationNode.getId().getOfficeId());
+            return null;
+        });
+    }
+
+    /**
+     * Delete a stream location
+     * @param locationId - the id of the stream location
+     * @param streamId - the id of the stream
+     * @param officeId - the office id
+     */
+    public void deleteStreamLocation(String officeId, String streamId, String locationId) {
+        connectionResult(dsl, conn -> {
+            CWMS_STREAM_PACKAGE.call_DELETE_STREAM_LOCATION(DSL.using(conn).configuration(), locationId, streamId, officeId);
+            return null;
+        });
+    }
+
+    static StreamLocation fromJooqStreamLocation(RETRIEVE_STREAM_LOCATION streamLocation, String locationId, String streamId, String officeId, String stationUnit, String stageUnit, String areaUnit) {
+        return new StreamLocation.Builder()
+                .withStreamLocationNode(buildStreamLocationNode(officeId,
+                        streamId,
+                        locationId,
+                        streamLocation.getP_PUBLISHED_STATION(),
+                        Bank.fromCode(streamLocation.getP_BANK()),
+                        stationUnit))
+                .withPublishedStation(streamLocation.getP_PUBLISHED_STATION())
+                .withNavigationStation(streamLocation.getP_NAVIGATION_STATION())
+                .withLowestMeasurableStage(streamLocation.getP_LOWEST_MEASURABLE_STAGE())
+                .withTotalDrainageArea(streamLocation.getP_DRAINAGE_AREA())
+                .withUngagedDrainageArea(streamLocation.getP_UNGAGED_DRAINAGE_AREA())
+                .withAreaUnits(areaUnit)
+                .withStageUnits(stageUnit)
+                .build();
+    }
+
+    static StreamLocationNode buildStreamLocationNode(String officeId, String streamId, String locationId, Double station, Bank bank, String stationUnits) {
+        return new StreamLocationNode.Builder()
+                .withId(new CwmsId.Builder()
+                        .withName(locationId)
+                        .withOfficeId(officeId)
+                        .build())
+                .withStreamNode(new StreamNode.Builder()
+                        .withStreamId(new CwmsId.Builder()
+                                .withName(streamId)
+                                .withOfficeId(officeId)
+                                .build())
+                        .withStation(station)
+                        .withBank(bank)
+                        .withStationUnits(stationUnits)
+                        .build())
+                .build();
+    }
+
+    static StreamLocation fromJooqStreamLocationRecord(Record record, String stationUnit, String stageUnit, String areaUnit) {
+        return new StreamLocation.Builder()
+                .withStreamLocationNode(buildStreamLocationNode(
+                        record.get(STREAM_LOCATION_OFFICE_ID_COLUMN_INDEX, String.class),
```

Comment:
I'm fairly certain you can pass in the field (you can manually define JOOQ fields) or the actual name.
Using the index not generally preferred as it is possible, though unlikely, for columns to move.

Reply:
Updated to pass in field name rather than index